### DISCUSSION TITLE
Can I install it in an existing wp plugin setup?

### DISCUSSION THREAD

**Discussion by michaelgrossklos** [dingo-d/wp-pest#33] [Q&A] [answered]
I want to use it in an already existing wp plugin dev setup. I use localwp and a normal wp installation. Is it possible or can I only use this on plugins to be setup?

**Comment by dingo-d** [comment]
I think it could be used, but you'd need to see how `bootstrap.php` file is set up and how it executes tests. I'm also using different base tests for running tests with Pest. So you'd need to check if they are correctly included in the bootstrap files.

You can check how the bootstrap file for plugin looks like in this lib: https://github.com/dingo-d/wp-pest/blob/main/templates/bootstrap-plugin.php.tmpl

**Reply by michaelgrossklos** [reply]
No, it does not work. Even though the `bootstrap.php` looks fine to me
I get the error:
![CleanShot 2023-12-13 at 20 47 00](https://github.com/dingo-d/wp-pest/assets/6610580/afc8d3aa-1cef-486f-9cbe-f4f2f2d23412)

```PHP
<?php

use Yoast\WPTestUtils\WPIntegration;

// Autoload everything for unit tests.
$ds = DIRECTORY_SEPARATOR;
require_once dirname(__FILE__, 2) . $ds . 'vendor' . $ds . 'autoload.php';

/**
 * Include core bootstrap for an integration test suite
 *
 * This will only work if you run the tests from the command line.
 * Running the tests from IDE such as PhpStorm will require you to
 * add additional argument to the test run command if you want to run
 * integration tests.
 */
if (isset($GLOBALS['argv']) && in_array('--group=integration', $GLOBALS['argv'], true)) {
    if (!file_exists(dirname(__FILE__, 2) . '/wp/tests/phpunit/wp-tests-config.php')) {
        // We need to set up core config details and test details
        copy(
            dirname(__FILE__, 2) . '/wp/wp-tests-config-sample.php',
            dirname(__FILE__, 2) . '/wp/tests/phpunit/wp-tests-config.php'
        );
        
        // Change certain constants from the test's config file.
        $testConfigPath = dirname(__FILE__, 2) . '/wp/tests/phpunit/wp-tests-config.php';
        $testConfigContents = file_get_contents($testConfigPath);
        
        $testConfigContents = str_replace(
            "dirname( __FILE__ ) . '/src/'",
            "dirname(__FILE__, 3) . '/src/'",
            $testConfigContents
        );
        $testConfigContents = str_replace("youremptytestdbnamehere", $_SERVER['DB_NAME'], $testConfigContents);
        $testConfigContents = str_replace("yourusernamehere", $_SERVER['DB_USER'], $testConfigContents);
        $testConfigContents = str_replace("yourpasswordhere", $_SERVER['DB_PASSWORD'], $testConfigContents);
        $testConfigContents = str_replace("localhost", $_SERVER['DB_HOST'], $testConfigContents);
        
        file_put_contents($testConfigPath, $testConfigContents);
    }
    
    // Give access to tests_add_filter() function.
    require_once dirname(__FILE__, 2) . '/wp/tests/phpunit/includes/functions.php';
    
    /**
     * Manually load the plugin being tested.
     */
    function _manually_load_plugin()
    {
        require dirname(__DIR__) . '/wpfaker.php';
    }
    
    tests_add_filter('muplugins_loaded', '_manually_load_plugin');
    
    require_once dirname(__DIR__) . '/vendor/yoast/wp-test-utils/src/WPIntegration/bootstrap-functions.php';
    
    /*
     * Bootstrap WordPress. This will also load the Composer autoload file, the PHPUnit Polyfills
     * and the custom autoloader for the TestCase and the mock object classes.
     */
    WPIntegration\bootstrap_it();
}
```

This is my file structure:
![CleanShot 2023-12-13 at 20 45 58](https://github.com/dingo-d/wp-pest/assets/6610580/c81a5e6f-4fd5-48b5-844f-4ccc5c0f5d2f)

**Reply by dingo-d** [reply]
How does your `phpunit.xml` look like? This looks like a standard installation of `wp-pest` (meaning the WordPress used will be the one in the `wp` folder with the SQLite database). You'd like to use the one that is set up using WPCLI if I understood correctly? And that setup uses tmp folder to install WordPress, and a mysql database for testing.

**Reply by michaelgrossklos** [reply]
Thanks for answering. You are correct. But I didn't explicitly use WPCLI It is a standard LocalWP by Flyweel installation and yes, I'd like to use that.

Here's my `phpunit.xml`

```XML
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
		 xsi:noNamespaceSchemaLocation="./vendor/phpunit/phpunit/phpunit.xsd"
		 bootstrap="tests/bootstrap.php"
		 colors="true"
>
	<testsuites>
		<testsuite name="Unit Test Suite">
			<directory>./tests/Unit/</directory>
		</testsuite>
		<testsuite name="Integration Test Suite">
			<directory>./tests/Integration/</directory>
		</testsuite>
	</testsuites>
	<php>
		<env name="TEST" value="true" force="true" />
		<env name="WP_TESTS_DIR" value="wp/tests/phpunit"/>
		<server name="DB_NAME" value="wp_pest_test_db"/>
		<server name="DB_USER" value="root"/>
		<server name="DB_PASSWORD" value=""/>
		<server name="DB_HOST" value="localhost"/>
	</php>
	<coverage processUncoveredFiles="true">
		<include>
			<directory suffix=".php">./src</directory>
		</include>
		<report>
			<clover outputFile="tests/coverage/clover.xml"/>
			<html outputDirectory="tests/coverage/html" lowUpperBound="50" highLowerBound="90"/>
		</report>
	</coverage>
</phpunit>
```

**Reply by dingo-d** [reply, marked as answer]
I think I misunderstood you. I thought you meant using this with the WPCLI test scaffold command 🤦🏼‍♂️ Sorry, my bad.

I mean, it should work on any setup, as long as you initialize it correctly. From what I can see it is set up correctly.

Did you try running `vendor/bin/pest --group=unit` to run the tests?

I haven't tested this with local by Flywheel so I'm not 100% sure what could be the issue. 

It should work on mac with Laravel Valet, or any local setups that have terminal (WSL or linux with locally set up dev server).

**Reply by michaelgrossklos** [reply]
`vendor/bin/pest --group=unit` was the command I used as I've encountered the error. Thanks for you help so far. I will look into that. Maybe I switch to another setup. I am not bound to flyweel.

**Reply by dingo-d** [reply]
Where did you run the command from? Because I know local is like a dockerized dev env, so you may need to SSH to the container in order to run the setup and the tests.

**Reply by michaelgrossklos** [reply]
Ah OK. I ran it just in terminal in my project without SSH into a container. Thanks for the tip.
